inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -0,0 +1,115 @@
+= Releasing INCEpTION
+
+== Prerequisites
+
+When releasing and publishing the resulting artifact, make sure that you have actually the rights
+to publish new artifacts to the target repository. See e.g.
+https://maven.apache.org/settings.html#Servers[the Maven documentationon authentication] or
+https://dkpro.github.io/dkpro-core/pages/setup-maven[DKPro Core - Setting up Maven repository]
+for more information on how to do this. Additionally, it assumes that you have write permissions
+to the protected `master` branch and to the protected maintenance branches.
+
+== Branching model
+
+INCEpTION uses a branching model where the development towards the next feature release happens on the
+`master`. Additionally, there are *maintenance* (also called *stable*) branches for all past feature 
```

Comment:
maybe add to the description of maintenance branches that they are called 0.6.x, 0.7.x , ...